framework/field_functions/interpolation/ffinter_line.cc

```diff
@@ -69,196 +60,132 @@ void
-
-    ff_ctx.interpolation_points_values.assign(number_of_points_, 0.0);
-    for (int p = 0; p < number_of_points_; ++p)
+  const auto& sdm = ref_ff_->GetSpatialDiscretization();
+  const auto& grid = sdm.Grid();
+  const auto& uk_man = ref_ff_->GetUnknownManager();
+  const auto uid = 0;
+  const auto cid = ref_component_;
+  const auto field_data = ref_ff_->GetGhostedFieldVector();
+
+  double local_max = 0.0, local_sum = 0.0, local_avg = 0.0;
+  size_t local_size = local_interpolation_points_.size();
+  local_interpolation_values_.resize(local_size);
+  for (auto p = 0; p < local_size; ++p)
+  {
+    auto& point = local_interpolation_points_[p];
+    auto cell_local_index = local_cells_[p];
+    const auto& cell = grid.local_cells[cell_local_index];
+    const auto& cell_mapping = sdm.GetCellMapping(cell);
+    const size_t num_nodes = cell_mapping.NumNodes();
+
+    std::vector<double> shape_function_vals(num_nodes, 0.0);
+    cell_mapping.ShapeValues(point, shape_function_vals);
+    double point_value = 0.0;
+    for (size_t i = 0; i < num_nodes; ++i)
     {
-      if (not ff_ctx.interpolation_points_has_ass_cell[p])
-        continue;
-
-      const auto cell_local_index = ff_ctx.interpolation_points_ass_cell[p];
-      const auto& cell = grid.local_cells[cell_local_index];
-      const auto& cell_mapping = sdm.GetCellMapping(cell);
-      const size_t num_nodes = cell_mapping.NumNodes();
-
-      std::vector<double> shape_function_vals(num_nodes, 0.0);
-      cell_mapping.ShapeValues(interpolation_points_[p], shape_function_vals);
-
-      double point_value = 0.0;
-      for (size_t i = 0; i < num_nodes; ++i)
-      {
-        const int64_t imap = sdm.MapDOFLocal(cell, i, uk_man, uid, cid);
+      const int64_t imap = sdm.MapDOFLocal(cell, i, uk_man, uid, cid);
+      point_value += shape_function_vals[i] * field_data[imap];
+    }
+    local_interpolation_values_[p] = point_value;
+    local_max = std::max(point_value, local_max);
+    local_sum += point_value;
+  }
 
-        point_value += shape_function_vals[i] * field_data[imap];
-      } // for node i
-      ff_ctx.interpolation_points_values[p] = point_value;
-    } // for p
-  }   // for ff
+  if (op_type_ == FieldFunctionInterpolationOperation::OP_SUM)
+    mpi_comm.all_reduce(local_sum, op_value_, mpi::op::sum<double>());
+  else if (op_type_ == FieldFunctionInterpolationOperation::OP_AVG)
+  {
+    size_t global_size = 0;
+    mpi_comm.all_reduce(local_size, global_size, mpi::op::sum<size_t>());
+    double global_sum = 0.0;
+    mpi_comm.all_reduce(local_sum, global_sum, mpi::op::sum<double>());
+    op_value_ = global_sum / static_cast<double>(global_size);
+  }
+  else if (op_type_ == FieldFunctionInterpolationOperation::OP_MAX)
+    mpi_comm.all_reduce(local_max, op_value_, mpi::op::max<double>());
 }
 
 void
-FieldFunctionInterpolationLine::ExportPython(std::string base_name)
+FieldFunctionInterpolationLine::Export(std::string base_name)
```

Comment:
Should this be called `ExportToCSV`?

Reply:
I debated that. What I was trying to prevent was having an `ExportToCSV`, `ExportToHDF`, `ExportToNNN` -  some of which may not applicable (or even do anything) for every field function. For example, you probably wouldn't want to try to define `ExportToCSV` for volume field functions. It turns out that volume doesn't have export functionality, point probably doesn't really need it, and that just leaves slice and line. Slice is likely going away, so that's just going to leave line, and I went with just plain `Export`. Happy to change it though.

Reply:
Fixed.